### PR TITLE
feat: Introduce JNA binding [AR-1246]

### PR DESCRIPTION
**Pull request by ohassine**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Some methods  from Avs library, that we have to use for calling, are implemented in C/C++.
These methods are not accessible directly from Kotlin

### Solutions

This PR aims to introduce JNA to bind C/C++ methods and make them accessible using kotlin

### Dependencies

- [x] #259

### Testing

Manually tested

----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: calling/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    }
+    android()
+
+    sourceSets {
+        val androidMain by getting {
+            dependencies {
+                implementation(Dependencies.Calling.avs)
+                implementation(Dependencies.Calling.jna)
+                implementation(Dependencies.Cryptography.cryptoboxAndroid)
```
Comment (typfel) [inline calling/build.gradle.kts:40]: Why do we have a dependency on `cryptoboxAndroid`?

**Review by typfel — CHANGES_REQUESTED** [review]

File: calling/build.gradle.kts (hunk 1 above)
Comment (ohassine) [reply calling/build.gradle.kts:40]: @typfel  libsodium is needed to use AVS

Comment (typfel) [reply calling/build.gradle.kts:40]: Libsodium is statically linked into the AVS library there no need to provide it at runtime.

**Review by typfel — CHANGES_REQUESTED** [review]
let's try and remove the cryptobox dependency.

**Pull request merged**